web-bundle/src/main/java/com/graphhopper/http/GraphHopperManaged.java

```diff
@@ -61,26 +69,33 @@ public GraphHopperManaged(CmdArgs configuration, ObjectMapper objectMapper) {
         } else {
             graphHopper = new GraphHopperOSM(landmarkSplittingFeatureCollection).forServer();
         }
-        String spatialRuleLocation = configuration.get("spatial_rules.location", "");
-        if (!spatialRuleLocation.isEmpty()) {
+        String spatialRuleBordersDirLocation = configuration.get("spatial_rules.borders_directory", "");
+        if (!spatialRuleBordersDirLocation.isEmpty()) {
             final BBox maxBounds = BBox.parseBBoxString(configuration.get("spatial_rules.max_bbox", "-180, 180, -90, 90"));
-            try (final InputStreamReader reader = new InputStreamReader(new FileInputStream(spatialRuleLocation), UTF_CS)) {
-                JsonFeatureCollection jsonFeatureCollection = localObjectMapper.readValue(reader, JsonFeatureCollection.class);
-                SpatialRuleLookupHelper.buildAndInjectSpatialRuleIntoGH(graphHopper, maxBounds, jsonFeatureCollection);
+            final Path bordersDirectory = Paths.get(spatialRuleBordersDirLocation);
+            List<JsonFeatureCollection> jsonFeatureCollections = new ArrayList<>();
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream(bordersDirectory, "*.{geojson,json}")) {
```

Comment:
This is nice! I can use this für #1776 too.

Reply:
Note that [DirectoryStream](https://developer.android.com/reference/java/nio/file/DirectoryStream) was added on Android in API level 26 (Android 8.0 Oreo). :slightly_smiling_face:

Reply:
It is not in the `core` module, so this should not be a problem.

Reply:
See https://github.com/graphhopper/graphhopper/pull/1841/files#diff-130ed9726000f7f80461f0c819f1e5c4R82

Reply:
But it seems there is no simple way to get the file name without file extension and additionally one has to create a Buffer from the Path, so I think I'll keep using the newly introduced Helper.listFiles method https://github.com/graphhopper/graphhopper/pull/1841/files#diff-4e518ca1e483cdb8f16fea150c2fb150R171 (this would also make this code here a few lines shorter due to the catch stuff)

Reply:
The correct handling of paths makes the `Path` API superior IMHO:
```java
Path folder = Paths.get("a", "b", "c");
Path file = folder.resolve("d");
```
No need to worry about OS specific stuff or if there's a missing `\` or `/`.